.gitignore

```diff
@@ -168,6 +168,7 @@ cmake-build-debug/
 /test/compflags/waynew/*.h
 /test/compflags/waynew/chplmake.out
 
+/test/deprecated/IO/test.txt
```

Comment:
For future reference, we're trying to move away from having `.gitignore` entries in this centralized file, preferring to put them into the directory in question instead (rationale being:  It makes it easier to move stuff around, and arguably easier when you're studying a given directory to see what is or is not ignored within it).  I don't think this is worth changing, but good to know going forward.

Reply:
I had initially put this in the directory level `.gitignore` but I was told that the convention was to put this in the top level `.gitignore`? I think people might be on different pages regarding this.

Reply:
Honestly, I could just have been misremembering.  Brad, do you know if we document that decision anywhere?

Reply:
> Brad, do you know if we document that decision anywhere?

I haven't done the `git grep`s to know for sure, but wouldn't be surprised if it were not documented.  That said, it seems like the kind of thing that, if documented, nobody would read until they did the other thing and had it pointed out to them (?).  A good but thankless housecleaning task would be to take the files currently in `.gitignore` and move them to directory-specific `.gitignore`s so that someone looking at it would only see local / global entries (and maybe a comment there stating that that's the approach we take).  That would serve as more of a self-documenting way of noting it (not that I'm opposed to documenting it as well...).